### PR TITLE
 Add Prometheus Remote Write Exporter supporting Cortex - factory and config

### PR DESCRIPTION
**Pull request by huyan0**
**Description:**
This PR replaces the close PR #1524. Comments on that PR has been addressed here. This PR is part one of  PR #1525. 

This PR implements the config, config YAML, and factory structs, the prwExporter struct. It also includes a sample configuration to run this Prometheus remote write exporter. The config and factory structs are fully compatible with the Collector component interfaces.

**Link to tracking Issue:** [#1150](https://github.com/open-telemetry/opentelemetry-collector/issues/1150)
 Related issues are:
 Metrics aggregation proposal: [#1422](https://github.com/open-telemetry/opentelemetry-collector/issues/1422)
 Prometheus exporter not functional: [#1255](https://github.com/open-telemetry/opentelemetry-collector/issues/1255)
 Related spec discussion: [#731](https://github.com/open-telemetry/opentelemetry-collector/pull/731)

**Documentation:**

* [Design](https://github.com/open-telemetry/opentelemetry-collector/blob/master/exporter/prometheusremotewriteexporter/DESIGN.md)

cc: [@huyan0](https://github.com/huyan0) [@alolita](https://github.com/alolita) [@markcartertm](https://github.com/markcartertm) [@sonofachamp](https://github.com/sonofachamp) [@morigs](https://github.com/morigs)
 Request for review: [@jmacd](https://github.com/jmacd) [@bogdandrutu](https://github.com/bogdandrutu) [@annanay25](https://github.com/annanay25) [@gouthamve](https://github.com/gouthamve) [@open-telemetry/collector-approvers](https://github.com/orgs/open-telemetry/teams/collector-approvers) [@open-telemetry/collector-maintainers](https://github.com/orgs/open-telemetry/teams/collector-maintainers)

### CODE REVIEW

**Comment by huyan0** [comment]
Regarding the comments on headers in PR #1524, I couldn't find a headers field inside HTTPClientSettings to merge our headers into; would appreciate some help on that part : )

**Comment by bogdandrutu** [comment]
I said that you should help us with that if you don't mind :). So please create a PR that adds Headers to the Http config :)

If you don't have time probably just file an issue and point to this file to resolve

**Comment by huyan0** [comment]
> I said that you should help us with that if you don't mind :). So please create a PR that adds Headers to the Http config :)
> 
> If you don't have time probably just file an issue and point to this file to resolve

Ah I see, thanks for the clarification. I will look into that and create a PR.

File: go.mod [hunk 1]
```diff
@@ -19,6 +19,7 @@ require (
 	github.com/gogo/protobuf v1.3.1
 	github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
 	github.com/golang/protobuf v1.4.2
+	github.com/golang/snappy v0.0.1
```
Comment (bogdandrutu) [inline go.mod:22]: Don't see where this is used yet. Probably add it later?

**Comment by huyan0** [comment]
Hi, I have updated the documents based on comments. I will leave the headers in config for now and remove them in the HTTPClientConfig PR.

File: exporter/prometheusremotewriteexporter/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+This Exporter sends metrics data in Prometheus TimeSeries format to Cortex or any Prometheus [remote write compatible backend](https://prometheus.io/docs/operating/integrations/).
+
+Non-cumulative monotonic, histogram, and summary OTLP metrics are dropped by this exporter. 
+
+The following settings are required:
+
+- `endpoint`: protocol:host:port to which the exporter is going to send traces or metrics, using 
+the HTTP/HTTPS protocol. 
+
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/README.md:11]: Either add empty lines between all or remove

File: go.sum [hunk 3]
```diff
@@ -10,6 +10,7 @@ cloud.google.com/go v0.50.0/go.mod h1:r9sluTvynVuxRIOHXQEHMFffphuXHOMZMycpNR5e6T
 cloud.google.com/go v0.51.0/go.mod h1:hWtGJ6gnXH+KgDv+V0zFGDvpi07n3z8ZNj3T1RW0Gcw=
 cloud.google.com/go v0.52.0/go.mod h1:pXajvRH/6o3+F9jDHZWQ5PbGhn+o8w9qiu/CffaVdO4=
 cloud.google.com/go v0.53.0/go.mod h1:fp/UouUEsRkN6ryDKNW/Upv/JBKnv6WDthjR6+vze6M=
```
Comment (bogdandrutu) [inline go.sum:12]: Go.sum changes from where?

**Pull request merged**